### PR TITLE
Chapter 12

### PR DESCRIPTION
**Pull request by mohdzakwan99**
Ready

### CODE REVIEW

**Comment by mohdzakwan99** [comment]
ready

File: app/assets/stylesheets/password_resets.scss [hunk 1]
```diff
@@ -0,0 +1,3 @@
+// Place all the styles related to the PasswordResets controller here.
```
Comment (anlvq-1058) [inline app/assets/stylesheets/password_resets.scss:1]: delete this file please

File: app/assets/stylesheets/static_pages.scss [hunk 2]
```diff
@@ -0,0 +1,3 @@
+// Place all the styles related to the StaticPages controller here.
```
Comment (anlvq-1058) [inline app/assets/stylesheets/static_pages.scss:1]: delete this file please

File: app/assets/stylesheets/users.scss [hunk 3]
```diff
@@ -0,0 +1,3 @@
+// Place all the styles related to the Users controller here.
```
Comment (anlvq-1058) [inline app/assets/stylesheets/users.scss:1]: delete this file please

File: app/controllers/password_resets_controller.rb [hunk 4]
```diff
@@ -0,0 +1,54 @@
+class PasswordResetsController < ApplicationController
+  before_action :get_user, only: [:edit, :update]
+  before_action :valid_user, only: [:edit, :update]
+  before_action :check_expiration, only: [:edit, :update]
```
Comment (anlvq-1058) [inline app/controllers/password_resets_controller.rb:4]: combine into 1 line

File: app/controllers/password_resets_controller.rb [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+      flash[:info] = "Email sent with password reset instructions"
+      redirect_to home_url
+    else
+      flash.now[:danger] = "Email address not found"
+      render :new
+    end
+  end
+
+  def edit;
+  end
```
Comment (anlvq-1058) [inline app/controllers/password_resets_controller.rb:19]:
```suggestion
  def edit; end
```

**Review by anlvq-1058 — COMMENTED** [review]
cmt

**Comment by mohdzakwan99** [comment]
ready

**Comment by ngocvt-0484** [comment]
merged

**Pull request merged**